kmeta/names_test.go

```diff
@@ -56,11 +58,25 @@ func TestChildName(t *testing.T) {
 		parent: "aaaa",
 		suffix: strings.Repeat("b---a", 20),
 		want:   "aaaa7a3f7966594e3f0849720eced8212c18b---ab---ab---ab---ab---ab",
-	}}
+	}, {
+		parent: strings.Repeat("a", 17),
+		suffix: "a.-.-.-.-.-.-.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+		want:   "aaaaaaaaaaaaaaaaaa1eb10dc911444f8434af83b7225442da",
+	}, {
+		parent: strings.Repeat("A", 17),
+		suffix: "A.-.-.-.-.-.-.AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+		want:   "aaaaaaaaaaaaaaaaae982551a43f641a37bf40c8feccd95f3a",
+	},
+	}
 
 	for _, test := range tests {
 		t.Run(test.parent+"-"+test.suffix, func(t *testing.T) {
-			if got, want := ChildName(test.parent, test.suffix), test.want; got != want {
+			got, want := ChildName(test.parent, test.suffix), test.want
+			errs := validation.IsDNS1123Subdomain(got)
+			if len(errs) != 0 {
```

Comment:
Try this (a standard Go pattern) for collapsing these two lines:

```suggestion
			if errs := validation.IsDNS1123Subdomain(got); len(errs) != 0 {
```

One thing to watch out for in this pattern is that moving the declaration of `errs` into the `if` statement makes `errs` scoped to the control block (the `if` statement). This can result in variable shadowing in some cases, though it's not relevant here.